2023/adv2023-rust/src/day11.rs

```diff
@@ -1,3 +1,115 @@
+use std::collections::{HashMap, HashSet};
+use std::io::{self, BufRead};
+// TODO: ^ Why BufRead if it's not used? What's self, just the io namespace?
```

Comment:
1) It brings `BufRead` trait into the scope. If you look at https://doc.rust-lang.org/std/io/trait.BufRead.html, it basically says that anything that implements `Read` trait, also now implements `BufRead` trait, which gives you all those methods, including `lines()`.
2) Yeah, that lets you refer to `std::io` as `io`. I think this is just a shorthand of `use std::io; use std::io::BufRead`.